### PR TITLE
CRNS-146/ts-thread

### PR DESCRIPTION
**Pull request by virdesai**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: src/components/Chat/hooks/useIsOnline.ts [hunk 1]
```diff
@@ -44,9 +41,9 @@ export const useIsOnline = <
 
     const handleRecoveredEvent = () => setConnectionRecovering(false);
 
-    const notifyChatClient = (netInfoState: NetInfoState) => {
-      if (client && client.wsConnection) {
-        if (netInfoState?.isConnected) {
+    const notifyChatClient = (netInfoState: boolean) => {
+      if (client?.wsConnection) {
+        if (netInfoState) {
```
Comment (nhannah) [inline src/components/Chat/hooks/useIsOnline.ts:46]: so something changed here?

File: src/components/Thread/__tests__/Thread.test.js [hunk 2]
```diff
@@ -84,7 +84,7 @@ describe('Thread', () => {
       <Chat client={chatClient}>
         <TranslationProvider value={{ ...translators, t }}>
           <Channel channel={channel} client={chatClient} thread={thread2}>
-            <Thread thread={thread2} />
+            <Thread />
```
Comment (nhannah) [inline src/components/Thread/__tests__/Thread.test.js:87]: Can we update thread in context between these renders? I think the point of the test here is to test the thread changing.

**Review by nhannah — APPROVED** [review]

File: src/components/Chat/hooks/useIsOnline.ts (hunk 1 above)
Comment (virdesai) [reply src/components/Chat/hooks/useIsOnline.ts:46]: yeah so this is passed in from the `native-package` stuff and then into `src/native.ts` and if you look at how it's returned there, it's sending back a boolean

File: src/components/Thread/__tests__/Thread.test.js (hunk 2 above)
Comment (virdesai) [reply src/components/Thread/__tests__/Thread.test.js:87]: yeah so passing thread values into `Channel` is what changes the context value of thread that's being used in Thread

**Pull request merged**